graph/list.go

```diff
@@ -70,13 +70,29 @@ func (s *TagStore) Images(config *ImagesConfig) ([]*types.Image, error) {
 	lookup := make(map[string]*types.Image)
 	s.Lock()
 	for repoName, repository := range s.Repositories {
+		filterTagName := ""
 		if config.Filter != "" {
-			if match, _ := path.Match(config.Filter, repoName); !match {
+			filterName := config.Filter
+			// Test if the tag was in there, if yes, get the name
+			if strings.Contains(filterName, ":") {
+				filterWithTag := strings.Split(config.Filter, ":")
+				filterName = filterWithTag[0]
+				filterTagName = filterWithTag[1]
+			}
+			if match, _ := path.Match(filterName, repoName); !match {
 				continue
 			}
+			if filterTagName != "" {
+				if _, ok := repository[filterTagName]; !ok {
+					continue
+				}
+			}
 		}
 		for ref, id := range repository {
 			imgRef := utils.ImageReference(repoName, ref)
+			if !strings.Contains(imgRef, filterTagName) {
```

Comment:
oh I didn't add a tests on that (trying with a part of an existing tag), might be worth to add (would document a bit more the behavior).